extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -37,6 +38,72 @@ class AmpStickyAd extends AMP.BaseElement {
       dev.warn(TAG, `TAG ${TAG} disabled`);
       return;
     }
+
+    const children = this.getRealChildren();
+    user.assert((children.length == 1 && children[0].tagName == 'AMP-AD'),
+        'amp-sticky-ad must have a single amp-ad child');
+    /** @const @private {!Element} */
+    this.ad_ = children[0];
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = this.getViewport();
+
+    /** @const @private {!Vsync} */
+    this.vsync_ = this.getVsync();
+
+    /** @const @private {!Unlisten}
+     * On viewport scroll, check requirements for amp-stick-ad to display.
+     */
+    this.onScrollListener_ =
```

Comment:
The returned function is not a listener, but instead a "unlisten" method. So, let's call `scrollUnlisten_` or such.